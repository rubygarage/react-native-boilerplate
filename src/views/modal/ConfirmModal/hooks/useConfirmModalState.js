import { useState } from 'react';
import { useDispatch } from 'react-redux';

const useConfirmModalState = (props) => {
  const { onCloseModal, onDestroyModal, confirmAction } = props;

  const dispatch = useDispatch();

  const [isShouldConfirm, setShouldConfirm] = useState(false);

  const handleShouldConfirm = () => {
    setShouldConfirm(true);
    onCloseModal();
  };

  const onModalHide = () => {
    if (isShouldConfirm) {
      dispatch(confirmAction);
    }
    onDestroyModal();
  };

  return {
    handleShouldConfirm,
    onModalHide,
    setShouldConfirm,
    isShouldConfirm,
  };
};

export default useConfirmModalState;
