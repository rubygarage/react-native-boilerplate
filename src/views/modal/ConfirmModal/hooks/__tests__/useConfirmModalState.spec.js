import { renderHook, act } from '@testing-library/react-hooks';

import useConfirmModalState from '../useConfirmModalState';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('useInputState custom hook', () => {
  const props = {
    onCloseModal: jest.fn(),
    onDestroyModal: jest.fn(),
    confirmAction: {},
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { result } = renderHook(() => useConfirmModalState(props));

    expect(result.current).toMatchSnapshot();
  });

  it('checks handleShouldConfirm method', () => {
    const { result } = renderHook(() => useConfirmModalState(props));

    act(() => {
      result.current.handleShouldConfirm();
    });

    expect(result.current.isShouldConfirm).toBe(true);
    expect(props.onCloseModal).toHaveBeenCalled();
  });

  describe('checks onModalHide method', () => {
    it('should dispatch and destroy modal', () => {
      const { result } = renderHook(() => useConfirmModalState(props));

      act(() => { result.current.setShouldConfirm(true); });

      act(() => { result.current.onModalHide(); });

      expect(mockDispatch).toHaveBeenCalledWith({});
      expect(props.onDestroyModal).toHaveBeenCalled();
    });

    it('should only destroy modal', () => {
      const { result } = renderHook(() => useConfirmModalState(props));

      act(() => { result.current.setShouldConfirm(false); });

      act(() => { result.current.onModalHide(); });

      expect(mockDispatch).not.toHaveBeenCalledWith({});
      expect(props.onDestroyModal).toHaveBeenCalled();
    });
  });
});
