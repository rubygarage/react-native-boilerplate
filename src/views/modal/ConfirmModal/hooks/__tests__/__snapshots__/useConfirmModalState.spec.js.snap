// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useInputState custom hook matches snapshot 1`] = `
Object {
  "handleShouldConfirm": [Function],
  "isShouldConfirm": false,
  "onModalHide": [Function],
  "setShouldConfirm": [Function],
}
`;
